src/component/form/Input.js

```diff
@@ -0,0 +1,40 @@
+import PropTypes from "prop-types";
+
+import {POSITION} from "../../tool/constant";
+
+import Tooltip from "../ui/Tooltip";
+
+import "./input.scss";
+
+const Input = ({type, placeholder, hasError, onChange}) => {
+
+    const inputRequiredError = () => {
+        return (
+            <div className="form__input-error__image__frame">
+                <img className="form__input-error__image" src="/images/error.svg" alt="input error"/>
+                <Tooltip text="Required" position={POSITION.RIGHT}/>
+            </div>
+        );
+    }
```

Comment:
We tend to refrain from following this sort of pattern, namely `Indirection`, as this doesn't make it clear what the flow of the markup is going to follow. Creating functions that render JSX generally means we should rather be creating a private component to render this markup